passkey: add bulk PRF support and simplify the passkey API surface (#909)
breez/spark-sdk: docs/breez-sdk/src/guide/passkey_prf_providers.md

# PRF providers

The built-in {{#name PasskeyProvider}} covers the common case. Reach for this page when:

> - You need fine-grained `PasskeyProvider` options (custom `userName`, `userDisplayName`, etc.).

Reviewer: ```suggestion
- You need fine-grained {{#name PasskeyProvider}} options (custom {{#name user_name}}, {{#name user_display_name}}, etc.).
```

- You're integrating Python, Go, or C# (no built-in `PasskeyProvider` ships for those bindings).
- You need a custom `PrfProvider` (CLI YubiKey, FIDO2, air-gapped backup file, hardware module).

## Built-in PasskeyProvider options

> In addition to `rpId` and `rpName`, the `PasskeyProvider` constructor accepts:

Reviewer: ```suggestion
In addition to {{#name rp_id}} and {{#name rp_name}}, the {{#name PasskeyProvider}} constructor accepts:
```

| {{#name user_name}} | {{#name rp_name}} | Maps to the WebAuthn `user.name`. Treated as the user's unique identifier for the credential and shown in the OS account picker during sign-in. Pass a stable per-user value if each registration should surface as a distinct entry (Apple's Passwords app, in particular, dedupes credentials by `(rpId, user.name)`; iOS additionally treats this as the only label exposed by `ASAuthorizationPlatformPublicKeyCredentialProvider`). Registration-only. |
| {{#name user_display_name}} | {{#name user_name}} | Maps to the WebAuthn `user.displayName`. The user-friendly label the OS / browser MAY (but is not required to) show in the picker; Android Credential Manager and the WebAuthn L3 picker surface it, iOS ignores. Registration-only. |

<div class="warning">
<h4>C# / Go / Python limitation</h4>

> The SDK does not ship a built-in `PasskeyProvider` for C#, Go, or Python (no native passkey API to wrap on those targets). Integrators on those bindings implement their own `PrfProvider` and pass it to `PasskeyClient::new(...)` directly.

Reviewer: `PasskeyClient::new(...)` is Rust syntax that reads as gibberish on other tabs. (Same for the bare `PasskeyProvider` / `PrfProvider` backticks at L6/L9/L26/L28 if you want those macro'd too.)

```suggestion
The SDK does not ship a built-in {{#name PasskeyProvider}} for C#, Go, or Python (no native passkey API to wrap on those targets). Integrators on those bindings implement their own {{#name PrfProvider}} and pass it to {{#name PasskeyClient}} directly.
```
